### ISSUE TITLE
Cache compcert

### ISSUE THREAD

**Issue by yforster** [palmskog/coq-program-verification-template#1]
It seems like most of the CI time is spent on building the dependencies. Is there a way to tell the CI to cache the files for e.g. CompCert in this GitHub workflow setup, or would this require a docker image with CompCert precompiled?

**Comment by palmskog** [comment]
@yforster yes, most of the CI time is spent building dependencies (VST, actually). The best way to handle this would likely be to build a dedicated Docker image based on Docker-Coq with the CompCert and VST dependencies included from the start. Erik Martin-Dorel has already done a lot of the automation groundwork for this with his [Docker-Base](https://github.com/coq-community/docker-base/blob/master/base/single/Dockerfile) and [Docker-Coq](https://github.com/coq-community/docker-coq/blob/master/coq/stable/Dockerfile) Docker files, and in particular with the [Docker Keeper tool](https://gitlab.com/erikmd/docker-keeper/-/wikis/home). See also the discussion in https://github.com/coq-community/docker-coq/issues/12.

**Comment by yforster** [comment]
Thanks, that's very helpful!

I'm mainly asking because we have very similar problems in the compilation chain MetaCoq -> [coq-library-undecidability](https://github.com/uds-psl/coq-library-undecidability/) -> [coq-library-complexity](https://github.com/uds-psl/coq-library-complexity).

I'll have a look whether we can get automation working to create docker images for the first two, or at least create docker images corresponding to opam releases.
